lib/indexer/comparer.rb

```diff
@@ -0,0 +1,122 @@
+# require_relative 'app';
+# require 'pry'
+# Indexer::Comparer.new(
+#   'government',
+#   'government-old-2017-06-21t11:04:56z-00000000-0000-0000-0000-000000000000',
+#   'government-2017-06-21t11:55:51z-00000000-0000-0000-0000-000000000000'
+# ).run
+
+module Indexer
+  class Comparer
+    # Source fields to ignore when comparing old and new objects for equality.
+    # These are ones which we know have changed for most documents.
+    FIELDS_TO_IGNORE = ["_id", "_type", "popularity"]
+
+    def initialize(base_name, old_index_name, new_index_name, rejected_fields = nil)
+      @base_name = base_name
+      @old_index_name = old_index_name
+      @new_index_name = new_index_name
+      @rejected_fields = rejected_fields || ['popularity']
+    end
+
+    def log(msg)
+      puts "#{Time.now}: #{msg}"
+    end
+
+    def read_index(index_name, filter = nil)
+      log "Access index: #{index_name}"
+      index = search_config.search_server.index_group(@base_name).index_for_name(index_name)
```

Comment:
I don't think this should use the `Index` class, since the old and new index name are passed in we can just instantiate the elasticsearch client directly.